alfresco-search/doc/architecture/event-trackers/event-based-content-tracker-spike.md

```diff
@@ -0,0 +1,298 @@
+
+After that, *Transform Engine* creates a new *Transform Reply* event (10) to be consumed by *Content Transformer*, where the content is retrieved from `Shared File Store` using `targetRef` (11).
+
+As the original request to `NodeContentGet` has been paused while performing the transformation, *Content Transformer* returns the TextContent Stream back and the response is delivered to SOLR (12).
+
+Both source and target files can be deleted from `Shared File Store` after the operation ends successfully (13,14).
+
+**Consequences**
+
+No consequences for *Search Services* are expected from applying this alternative. Probably then only one is that the whole indexing process will become slower, as the transformation is performed by the new Transform Service and contents are uploaded and downloaded using HTTP protocol. Anyway, this problem can be overcome by providing the right resources to Transform Service.
+
+
+### 2 - New Microservice in the Middle
+
+Introduce a new microservice that sits between Search Service and the Transformation Service. The content is off loaded to Transformation Service and the microservice stores the content for Search Service to ingest.
+
+![New Microservice in the Middle](content-tracker-2.png)
+
+**Components referenced in the figure**
+
+* [alfresco-search](https://github.com/Alfresco/SearchServices/tree/master/alfresco-search): Alfresco SOLR Application, deployed in SOLR Jetty web application `solr.war` as library
+* *DockerSOLR*: New REST API Layer based in Spring Boot App (to be dockerizable) to be designed. This component is also publishing `TransformRequest` events to be consumed by *Transform Service*
+* [alfresco-remote-api](https://github.com/Alfresco/alfresco-remote-api): REST API Layer for Alfresco Content Services, deployed in Alfresco Tomcat web application `alfresco.war` as library
+* [Shared File Store](https://github.com/Alfresco/alfresco-shared-file-store): REST API Layer to share contents between a *T-Client* and the *T-Engine*
+* [T-Service](https://github.com/Alfresco/alfresco-transform-service): Alfresco Transform Service to route JMS Events to T-Engines
+* [DockerTika](https://git.alfresco.com/Repository/alfresco-docker-transformers/tree/master/alfresco-docker-tika) REST API Layer for *T-Engine* based in Tika
+* [DockerLibreoffice](https://git.alfresco.com/Repository/alfresco-docker-transformers/tree/master/alfresco-docker-libreoffice) REST API Layer for *T-Engine* based in LibreOffice
+
+**Transformation Process**
+
+>> A new microservice, named *DockerSOLR* in the diagram, will handle transformations to text requests from Content Tracker. This service replaces previous `NodeContentGet` web script.
+
+Content Tracker invokes to *Docker SOLR* using HTTP REST API to start the transformation (1).
+
+*Docker SOLR* invokes the repository to retrieve the source content for the node using the Remote API (2) and stores the result in the `Shared File Store` to retrieve the `sourceRef` (3).
```

Comment:
Why would we need to call repo if we are listening to the que?

Reply:
In this alternative there is no one in the repository producing events for Search Services to be consumed. We are just only changing the *getTextContent* to use the new Transform Services. This is why we need to retrieve the original file from repo, as Transform Service requires this file stored in Shared File Store before attempting the transformation.